pym/portage/package/ebuild/doebuild.py

```diff
@@ -518,6 +526,23 @@ def doebuild_environment(myebuild, mydo, myroot=None, settings=None,
 				mysettings["KV"] = ""
 			mysettings.backup_changes("KV")
 
+	binpkg_compression = mysettings.get("BINPKG_COMPRESSION", "bzip2")
+	try:
+		compression = _compressors[binpkg_compression]
+	except KeyError as e:
+		raise TypeError("Invalid or unsupported compression method: %s" % e.args[0])
```

Comment:
If it's possible for the user to trigger this somehow, then we need to handle the exception somewhere and make emerge bail out before dependency calculations.

The BINPKG_COMPRESSION code should all go under the `if mydo != "depend":` conditional, since we want to minimize the code executed for the "depend" phase.